core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -742,7 +742,7 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
         longArray.set(pos * 2 + 1, keyHashcode);
         isDefined = true;
 
-        if (numKeys > growthThreshold && longArray.size() < MAX_CAPACITY) {
+        if (numKeys >= growthThreshold && longArray.size() < MAX_CAPACITY) {
```

Comment:
Unfortunately, we can't grow in the beginning, otherwise the `pos` will be wrong.